headers/types/common/common.h

```diff
@@ -992,6 +992,140 @@ struct wavi_data {
 };
 ASSERT_SIZE(struct wavi_data, 20); // Likely longer
 
+// Contains information about a running thread
+struct thread {
+    int flags;                // Probably a flags field. Usually 0x1F.
+    struct thread* field_0x4; // r2 parameter in ThreadStart
+    undefined4 field_0x8;     // Initialized to 0
+    undefined4 field_0xC;     // Initialized to 0
+    undefined4 field_0x10;    // Initialized to 0
+    undefined4 field_0x14;    // Initialized to 0
+    undefined4 field_0x18;    // Initialized to 0
+    undefined4 field_0x1C;    // Initialized to 0
+    undefined4 field_0x20;    // Initialized to 0
+    undefined4 field_0x24;    // Initialized to 0
+    undefined4 field_0x28;    // Initialized to 0
+    undefined4 field_0x2C;    // Initialized to 0
+    undefined4 field_0x30;    // Initialized to 0
+    undefined4 field_0x34;    // Initialized to 0
+    // 0x38: Points to the area of the allocated stack area that can actually be used to store data.
+    // This is equal to stack_pointer - 0x44, with an additional - 4 if the resulting addess is not
+    // aligned to 8 bytes.
+    void* usable_stack_pointer;
+    // 0x3C: Address of the function to return to once the thread exits. Usually ThreadExit.
+    void* return_address;
```

Comment:
Is this always a function with no params? Can we define a function pointer type?

```c
typedef void (*thread_exit_fn_t)(void);
```

Reply:
I didn't check. But I don't see a way to pass parameters, so probably?
Where would that definition be placed?

Reply:
Probably in `common.h` with all the other threading stuff. Before `struct thread` maybe?